addons/account/views/account_reconcile_model_views.xml

```diff
@@ -186,21 +189,26 @@
 
                                             <field name="sequence"
                                                    widget="handle"/>
-                                            <field name="account_id"/>
+                                            <field name="account_id"
+                                                   required="(parent.counterpart_type == 'general' and parent.rule_type == 'writeoff_button')
+                                                             or parent.rule_type=='writeoff_suggestion'"/>
                                             <field name="amount_type"/>
                                             <field name="journal_id"
                                                    column_invisible="parent.rule_type != 'writeoff_button'"
-                                                   optional="hide"/>
+                                                   optional="hide"
+                                                   domain="[('type', '=', self.counterpart_type)]"/>
```

Comment:
```suggestion
                                                   domain="[('type', '=', parent.counterpart_type)]"/>
```
I currently have a traceback when trying to set a journal on the reco model line. I think this solves the issue